general/longpages.py

```diff
@@ -84,15 +81,15 @@
                      FROM revision
                      WHERE rev_page = page_id)
 ORDER BY page_len, page_namespace ASC;
-''' , config.get('dbreps', 'dbname'))
+''')
 
 i = 1
 output1 = []
 for row in cursor.fetchall():
-    ns_name = u'%s' % unicode(row[0], 'utf-8')
+    page_namespace = u'{{subst:ns:%s}}' % unicode(row[0], 'utf-8')
     page_title = u'%s' % unicode(row[1], 'utf-8')
-    if ns_name:
-        page_title = '{{dbr link|1=%s:%s}}' % (ns_name, page_title)
+    if page_namespace:
+        page_title = '{{dbr link|1={{subst:ns:%s}}:%s}}' % (page_namespace, page_title)
     else:
         page_title = '{{dbr link|1=%s}}' % (page_title)
```

Comment:
Skimming this code, it seems unlikely that this else branch is reachable. I think `if page_namespace:` (and `if ns_name:` before it) will always evaluate as true. We're unconditionally casting to a string when establishing the `page_namespace` variable, as far as I can tell. This behavior is mostly unrelated to this commit, I just happened to notice due to the merge request.